roles/ceph-osd/tasks/docker/start_docker_osd.yml

```diff
@@ -40,17 +40,24 @@
     - ceph_osd_docker_prepare_env is defined
     - not osd_containerized_deployment_with_kv
 
+# use shell rather than docker module
+# to ensure osd disk prepare finishes before
+# starting the next task
 - name: prepare ceph osd disk with kv_store
-  docker:
-    image: "{{ ceph_osd_docker_username }}/{{ ceph_osd_docker_imagename }}"
-    name: "{{ ansible_hostname }}-osd-prepare-{{ item | regex_replace('/', '') }}"
-    net: host
-    pid: host
-    state: running
-    privileged: yes
-    volumes: "{{ item.0 }}:{{ item.0 }}"
-    env: "OSD_DEVICE={{ item.0 }},KV_TYPE={{kv_type}},KV_IP={{kv_endpoint}},{{ ceph_osd_docker_prepare_env }}"
-  ignore_errors: true
+  shell: |
```

Comment:
@leseb i just used convention which was present in this role. I can change it to command if you want.